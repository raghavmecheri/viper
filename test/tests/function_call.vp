int func test(int x) {
    return x * x;
}

int func sumOfList(int[] l) {
    int sum = 0;
    int i;
    for (i = 0; i < len(l); i+=1) {
        sum += l[i];
    }
    return sum;
}

bool func isIndexOfGreatest(int index, int[] l) {
    if (len(l) == 0) return false;
    int i;
    for (i = 0; i < len(l); i+=1) {
        if (l[i] > l[index]) {
            return false;
        }
    }
    return true;
}

print(test(10));

int[] fib = [1, 1, 2, 3, 5, 8, 13, 21];
print(sumOfList(fib));

int[] t = [1, 2, 3, 4, 5, 6, 7, 8];
print(isIndexOfGreatest(7, t));
print(isIndexOfGreatest(2, t));