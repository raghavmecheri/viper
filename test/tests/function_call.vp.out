100
54
1
0
